### DISCUSSION TITLE
Capture desktop audio

### DISCUSSION THREAD

**Discussion by Fractal-Tess** [libcala/wavy#51] [Q&A]
Hey there,

Is there a way to capture desktop audio on Linux using wavy?
I've been searching for some time, and I've looked through cpal, sfml and rodio, but as far as I can see, they do not seem to support this functionality.

**Comment by AldaronLau** [comment]
Hello,

Desktop audio capture on Linux works by configuring a virtual microphone.  As far as I know, there's no way for a library like wavy or cpal to do this configuration in a cross-platform way.  I've set it up using PulseAudio's pavuctl before, and then after setting it up, the desktop audio output shows up as a microphone in wavy.

**Reply by Fractal-Tess** [reply]
Is that also how open broadcast software works? I've always wondered what goes under the hood, but I feel like it is needlessly complicated for some reason. Still I would love it you were to give me a few pointers on where to look in order to learn more about this. I'll for sure check out pavuctl and what I can do with it.

**Reply by AldaronLau** [reply]
I Just did some research, and I think OBS talks to PulseAudio, JACK and Pipewire separately on Linux, which all have their own methods of exposing a loopback device.  None of which are guaranteed to be installed on any given Linux machine.

For a portable solution, you should be able to set up ALSA loopback devices with this command ([source](https://github.com/TheSalarKhan/Linux-Audio-Loopback-Device)):

```shell
sudo modprobe snd-aloop
```

Also [this stackoverflow post](https://unix.stackexchange.com/questions/12541/can-i-setup-a-loopback-audio-device) I found has a little more information about audio loopback devices on Linux in the answers.

**Reply by Fractal-Tess** [reply]
That makes a lot of sense.

I actually succeeded in capturing desktop audio with the help of pavucontrol.
I haven't dug deeper into the rabbit hole, but it all seems very magical to me and I still do not understand what exactly is going on under the hood. 
I'll try looking into it a bit more since I'm also trying to capture specific output streams as well.
